### PR TITLE
Heic support

### PR DESCRIPTION
**Pull request by Alexays**
Hi!

Using it for a while, no problem so far.
ported from https://github.com/ianare/exif-py

### CODE REVIEW

**Comment by bigflood** [comment]
@Alexays thanks

Would you upload sample file(s) and expected exif outputs?  
(so I can test it)

**Comment by Alexays** [comment]
Sorry for the delay @bigflood 
Here's a sample zipped.
You should get data like
```
Make: Apple
Model: iPhone X
```
[xje8hvTgojAaQheoJMnSTW.heic.zip](https://github.com/bigflood/dartexif/files/6513211/xje8hvTgojAaQheoJMnSTW.heic.zip)

File: lib/src/heic.dart [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    }
+  }
+
+  List<int> findExif() {
+    final ftyp = expectParse('ftyp');
+    assert(listEqual(ftyp.majorBrand, Uint8List.fromList('heic'.codeUnits)));
+    assert(ftyp.minorVersion == 0);
+    final meta = expectParse('meta');
+    assert(meta.subs['iinf']!.exifInfe != null);
+    final itemId = meta.subs['iinf']!.exifInfe.item_id;
```
[the remark this answers is not in the document]
Comment (Alexays) [reply lib/src/heic.dart:257]: item_id is part of exifInfe that is parsed from bytes, not code related.
However, the variable is named itemId

[the remark this answers is not in the document]
Comment (bigflood) [reply lib/src/heic.dart:257]: NoSuchMethodError has occured while parsing sample file.

```
$ dart ./bin/print_exif.dart ../exif/xje8hvTgojAaQheoJMnSTW.heic 
Opening: ../exif/xje8hvTgojAaQheoJMnSTW.heic
Unhandled exception:
NoSuchMethodError: Class 'HeicBox' has no instance getter 'item_id'.
Receiver: Instance of 'HeicBox'
Tried calling: item_id
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
#1      HEICExifFinder.findExif (package:exif/src/heic.dart:257:48)
#2      readExifFromFileReader (package:exif/src/read_exif.dart:82:22)
#3      readExifFromBytes (package:exif/src/read_exif.dart:25:10)
#4      printExifOfBytes (package:exif/src/print_exif.dart:9:22)
#5      main (file:///home/flood/git/dartexif/bin/print_exif.dart:62:17)
#6      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:281:32)
#7      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
```

[the remark this answers is not in the document]
Comment (Alexays) [reply lib/src/heic.dart]: Ok got it, my bad, just renamed the getter :)
Thanks!

**Comment by bigflood** [comment]
@Alexays 
Can I commit the sample file into github and use it as test data?

this file -> xje8hvTgojAaQheoJMnSTW.heic

**Comment by Alexays** [comment]
Sure @bigflood!

**Pull request merged**